langchain/vectorstores/elastic_vector_search.py

```diff
@@ -181,7 +181,7 @@ def add_texts(
 
         for i, text in enumerate(texts):
             metadata = metadatas[i] if metadatas else {}
-            _id = str(uuid.uuid4())
+            _id = str(metadata.pop("_id", uuid.uuid4())
```

Comment:
thoughts on instead adding `ids: Optional[List[str]] = None` as a kwarg to method? that's been convention with other vectorstores

Reply:
I think you mean ```ids: Optional[Sequence[str]] = None``` :wink:  

@dev2049 providing the ids to the method as part of the function signature does make a lot of sense to me.

Reply:
@dev2049 indeed you are right. Apply similiar solution to elastic: #5190